RecoLocalTracker/SiStripZeroSuppression/src/SiStripAPVRestorer.cc

```diff
@@ -976,79 +942,280 @@ bool SiStripAPVRestorer::CheckBaseline(const std::vector<int16_t> &baseline) con
+        isMax=true;
+        isAuxiliary_Minimum=true;//for minimums after strip 127-SW-1
+      }
+
+
+      if (!discontinuities.empty()){
+        itdiscontinuities=discontinuities.end();
+        --itdiscontinuities;
+      }
+
+      if ((isMax==true)&&(actualStripADC<=first_start_cluster_ADC)) {
+
+        if ((abs(first_start_cluster_ADC - singleAPVdigi[strip+2])>first_gradient)&&(isPossible_wrong_minimum==true)) {
+          discontinuities.erase(itdiscontinuities);
+          discontinuities.insert(discontinuities.end(), std::pair<int, int >(strip_first_gradient, ADC_start_point_cluster_pw));
+          discontinuities.insert(discontinuities.end(), std::pair<int, int >(strip, ADC_start_point_cluster_pw));//????
+        } else {
+          if ((discontinuities.size()%2==0)&&(discontinuities.size()>1)){
+            itdiscontinuities=discontinuities.end();
+            --itdiscontinuities;
+            discontinuities.erase(itdiscontinuities);
+          }
+          discontinuities.insert(discontinuities.end(), std::pair<int, int >(strip, actualStripADC));
+        }
+        isMax=false;
+        ADC_start_point_cluster_pw=0;
+        isPossible_wrong_minimum=false;
+        strip_first_gradient=0;
+        first_start_cluster_strip=0;
+        first_start_cluster_ADC=0;
+      }
+
+      if ((isMax==true)&&((actualStripADC-previousStripADC)>=0)&&(isAuxiliary_Minimum==false)){     //For the end Poit when strip >127-SW-1
+        if ((abs(first_start_cluster_ADC - singleAPVdigi[strip+1])>first_gradient)&&(isPossible_wrong_minimum==true)) {
+          discontinuities.erase(itdiscontinuities);
+          discontinuities.insert(discontinuities.end(), std::pair<int, int >(strip_first_gradient, ADC_start_point_cluster_pw));
+        }
+
+        if ((discontinuities.size()%2==0)&&(discontinuities.size()>1)){
+          itdiscontinuities=discontinuities.end();
+          --itdiscontinuities;
+          discontinuities.erase(itdiscontinuities);
+        }
+        discontinuities.insert(discontinuities.end(), std::pair<int, int >(strip-1, previousStripADC));
+        isMax=false;
+        ADC_start_point_cluster_pw=0;
+        isPossible_wrong_minimum=false;
+        strip_first_gradient=0;
+        first_start_cluster_strip=0;
+        first_start_cluster_ADC=0;
+      }
     }
-    return apvFlagsBool_;	
-}
+  }
+
+  //----------THIRD PART reconstruction of the event without baseline-------//
 
+  if(!discontinuities.empty()){
+    if ((first_start_cluster_strip)==127-1) discontinuities.insert(discontinuities.end(), std::pair<int, int >(127, first_start_cluster_ADC));
+    if ((isMax==true)&&(isAuxiliary_Minimum==true)) discontinuities.insert(discontinuities.end(), std::pair<int, int >(auxiliary_end_cluster, first_start_cluster_ADC));
+  }
 
+  if (isFirstStrip==true){
+    itdiscontinuities=discontinuities.begin();
+    ++itdiscontinuities;
+    ++itdiscontinuities;
+    //int firstStrip= itdiscontinuities->first;
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant